sycl/doc/extensions/PropertyList/SYCL_EXT_ONEAPI_property_list.asciidoc

```diff
@@ -264,72 +279,39 @@ a|
 using value_t = first;
 ```
 | The parameter type. Available only when there is exactly one parameter.
```

Comment:
I know you didn't change this line, but I think the specification of `value_t` needs to be clarified.  Let's take the "baz" example from above:

```
// This property has one integer non-type parameter.
struct baz_key {
  template<int K>
  using value_t = property_value<baz, std::integral_constant<int, K> >;
  // Note: integral_constant is used for the example. An implementation can use a different mapping.
};
// baz is an object of a property value type of baz.
template<int K>
inline constexpr baz_key::value_t<K> baz;
```

What is `property_value::value_t` in this case?  Is it `int` or `std::integral_constant<int, K>`?  I was assuming it was `int` because that seems more useful.  However, the example code above `using value_t = first` would result in `std::integral_constant<int, K>`.

If we really do intend `value_t` to be `std::integral_constant<int, K>`, then I think we need to change the description to say something like:

> Available only when there is exactly one parameter.  An unspecified implementation defined type that represents the property parameter.

However, I don't see why this is useful to applications.

Reply:
I don't see how `int` would be useful. Yes the intention is it is the literal type representing the value. It is useful if the user wants to do something with propert values independent of whether it holds a value or type. The user only needs to know that the type has a `::value` member which is guranteed by the spec.

Reply:
I don't see how it could be useful for the alias to be `std::integral_constant<int, K>` because we said that is an implementation detail.  Some other (non DPC++) implementation might represent the value in some other way.  Therefore, I don't see how an application can portably use `value_t` in this case.

Perhaps the best solution is to say that `value_t` is only available when the parameter is a type (i.e. not a non-type parameter).  In that case, it would provide the type of the parameter, which would be portable.  The table would then look something like this:

````
--
[options="header"]
|====
| Member | Description
a|
```c++
static constexpr auto value = /* unspecified */;
```
| The value of the parameter. Available only when there is exactly one parameter
whose value is a non-type.
a|
```c++
using value_t = /* unspecified */;
```
| The type of the parameter. Available only when there is exactly one parameter
whose value is a type.
a|
```c++
using key_t = Property;
```
| The property key type.
|===
--
````

Reply:
You don't need to know what exact type it is to check whether is matches (`is_same`) or to convert it to value. I think it is useful therefore I don't want to remove it. If you can think of uses which are prohibited by making it impl.-defined. I suggest we require it to be `std::integral_constant`.

Reply:
I agree that you could use `is_same` to check for matches.  If you think it's useful, let's just clarify the documentation.  How about:

````
a|
```c++
using value_t = First;
```
| Available only when there is exactly one parameter.  When the parameter's value is a type, `value_t`
is that type.  When the parameter's value is a non-type, `value_t` is an implementation-defined type
that represents the non-type value.
````